lib/vrl/parser/src/ast.rs

```diff
@@ -1077,6 +1077,31 @@ impl fmt::Debug for Not {
     }
 }
 
+// -----------------------------------------------------------------------------
+// abort
+// -----------------------------------------------------------------------------
+
+#[derive(Clone, PartialEq)]
+pub struct Abort {
+    pub message: Option<Box<Node<Expr>>>,
+}
+
+impl fmt::Display for Abort {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if let Some(message) = self.message.as_ref() {
+            write!(f, "abort: {}", message)
+        } else {
+            write!(f, "abort")
+        }
```

Comment:
nit:

```suggestion
        f.write_str(self.message.as_ref()
            .map(|m| format!("abort: {}", m))
            .unwrap_or_else(|| "abort".to_owned()))
```

(but I'll concede that I can't say either is "better")